### ISSUE TITLE
Bring docs into core repo

### ISSUE THREAD

**Issue by SachaG** [VulcanJS/Vulcan#1919]
Having the docs be part of the core repo would encourage PRs. We could even run them as part of Vulcan itself in their own `vulcan:docs` package to make them available offline and avoid the need for a separate Hexo stack.

**Comment by eric-burel** [comment]
About rendering markdown, I've tried [this in Awesome Vulcan for the home page](https://github.com/lbke/awesome-vulcan/blob/develop/packages/core/lib/components/Home.jsx#L51).
I managed to get SSR only by wrapping the markdown and export it as a string in a js file, since as you said there is no .md parser in Meteor. I lost i18n in the process. Globally not very satisfying...

- You can publish the markdown as a public asset but then server side you can't load it. It would works but without SSR. I've tried and the load time is very noticeable.

- Using a collection is a good idea, though it sounds more palliative since you use the most dynamic option to implement the most static feature. You might lose SSR too and have unexpected load time.

- Creating a Markdown "parser" for Meteor could be a relevant solution. It could consists in wrapping the .md text with a .js export automatically in order to produce a string. Then we would need to manage i18n somehow (eg importing a folder with all languages instead of a single file?).

I absolutely don't know how other frameworks like Gatsby handle this. I guess they literally build the HTML upfront by rendering the Markdown using React?

**Comment by Apollinaire** [comment]
For i18n in Awesome-Vulcan, since you are only exporting a string const maybe you can use `addString` from the intl package, add different translations, and get it back with `formatMessage` ?

**Comment by eric-burel** [comment]
Didn't think about this, that's quite innovative and indeed might work! I'll try that in a future release. And would it be a solution for the creation of a docs package?

**Comment by Apollinaire** [comment]
It will probably work, but is it worth it ?
I would find it a bit painful to edit some markdown into a string in a JS file (and escape all the `` ` ``!)

**Comment by eric-burel** [comment]
That's what I mean with by creating a loader for markdown, you could do this automatically during import, as you would do when using webpack. That's the only part of this solution that does not exists currently, but it may not be that hard to implement.

**Comment by Apollinaire** [comment]
I have a proof of concept for a mdx loader: https://github.com/Apollinaire/meteor-mdx
It compiles .mdx (.md + jsx) files to export a react component. So I think we could build the docs with dynamic loading of the content in mdx files. The bonus is that we could directly import JS in it just like in any jsx file. This is what Next.js did for their doc.
The caveat is that the extension has to be `.mdx` or Meteor will bundle every `.md` file, so every readme from the `node_modules` for example.
It works on a regular meteor app.... but not with a vulcan app. I don't understand why. If you can test it I would be glad. 
I managed to make it work with this repo: https://github.com/CaptainN/meteor-react-starter like this:

```sh
meteor add apollinaire:mdx
```
Just create a `myFile.mdx` anywhere in `imports` and then add: 
```js
import MyFile from `./myFile.mdx`;

<MyFile />
```

**Comment by Apollinaire** [comment]
Update : it works in vulcan, however it does not work **within a package**.
How I made this work: In Vulcan-starter : add a folder `server` , inside add `main.js` and `myFile.mdx`.
`main.jsx`:
```js
import Something from './myFile.mdx';
console.log(Something);
```

**Comment by SachaG** [comment]
Cool! We'd still want to publish the docs online though I think? It'd be great to think of a way to have the same content available as a "dynamic" copy as part of your Vulcan app and a "static" copy online. 

Maybe the "static" copy could be generated and deployed through a command line script when Vulcan is running locally? Kinda turning Vulcan into a static site generator in a way…

**Comment by Apollinaire** [comment]
Instead of trying to create a static site, why not just deploy a dynamic one? We could use the docs directly as a way to show the features of Vulcan. Or maybe I'm not getting your point with the static site

**Comment by SachaG** [comment]
Just because it's a pain to maintain a server, db, etc. Static sites are easier, you can deploy them and forget them.

**Comment by Apollinaire** [comment]
Ok, I see your point. 
So, the best thing to do is the following IMO:

- add the content in a special 'vulcan-docs' package in the main repo, that would also create the pages to your local build (maybe at `/vulcan-docs/nameofthepage`) if you `meteor add` it to your packages. And it could create a menu on our future debug bar/widget
- also add a `docs` folder that is ignored by Meteor and contains a gatsby or next site that gets its content from the mdx files (there is a plugin already) that are inside the 'docs-content' package

**Comment by Apollinaire** [comment]
Update about the loader: you can actually make it work from within a package by adding `apollinaire:mdx` to the `Package.onUse()` inside the `package.js` file. Also, I've published it to atmosphere

**Comment by SachaG** [comment]
Very cool. Let's create a new `docs` branch with a proof of concept, and then we can start working on this then. I think it'd be a good chance to rethink/rewrite the docs from scratch actually.

**Comment by Apollinaire** [comment]
I still need a bit of work on the loader to use caching and I'll try to get a POC working during my holiday ;) I'll be on it next week unless someone else starts it before

**Comment by Apollinaire** [comment]
I've created the `vulcan:docs` pacakges here: https://github.com/Apollinaire/Vulcan/tree/docsV2. It still needs polishing, but the pages are rendered. I need to integrate the mdx content to Gastby now, and we'll have a POC

**Comment by SachaG** [comment]
Oh awesome! Any idea how he could handle "dynamic" content like inserting your settings, resolvers, etc. directly inside the docs? Seems doable in Vulcan, but then we need some way of handling a fallback in Gatsby?

Maybe we could have a single large global object that contains everything the docs needs in terms of app-specific data, and then test for its presence in both context?

**Comment by Apollinaire** [comment]
```js
if(Meteor === undefined) {
  gatsby.doStuff();
} else {
  Vulcan.doStuff()
}
```
That should do it? Remember it's React components in the end!

**Comment by SachaG** [comment]
Yeah good point. As long as we keep that switch in a central place I think it'll be manageable. We just need to avoid having those `if(...)` in every component or MDX file.
